internal/painter/drawer.go

```diff
@@ -0,0 +1,86 @@
+package painter
+
+import (
+	"image"
+	"image/draw"
+	"log"
+	"math"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+	"golang.org/x/image/font"
+	"golang.org/x/image/math/fixed"
+)
+
+// Drawer extends "golang.org/x/image/font" to add support for tabs
+// Drawer draws text on a destination image.
+//
+// A Drawer is not safe for concurrent use by multiple goroutines, since its
+// Face is not.
+type Drawer struct {
+	font.Drawer
+}
+
+func tabStop(f font.Face, x fixed.Int26_6) fixed.Int26_6 {
+	spacew, ok := f.GlyphAdvance(' ')
+	if !ok {
+		log.Print("Failed to find space width for tab")
+		return x
+	}
+	tabWidth := fyne.CurrentApp().Settings().Theme().Size(theme.SizeNameTabWidth)
+	tabw := spacew * fixed.Int26_6(tabWidth)
+	tabs, _ := math.Modf(float64((x + tabw) / tabw))
+	return tabw * fixed.Int26_6(tabs)
+}
+
+// DrawString draws s at the dot and advances the dot's location.
+// Tabs are translated into a dot location change.
+func (d *Drawer) DrawString(s string) {
+	prevC := rune(-1)
+	for _, c := range s {
+		if prevC >= 0 {
+			d.Dot.X += d.Face.Kern(prevC, c)
+		}
+		if c == '\t' {
+			d.Dot.X = tabStop(d.Face, d.Dot.X)
+		} else {
+			dr, mask, maskp, a, ok := d.Face.Glyph(d.Dot, c)
+			if !ok {
+				// TODO: is falling back on the U+FFFD glyph the responsibility of
```

Comment:
I assume you mean the TODO?
I haven't a clue, I was quite surprised when I opened [font.go](https://github.com/golang/image/blob/master/font/font.go) for the first time and saw the number of TODOs. It has been like it for 6 years and it does not seem to be an issue so left it well alone.
It does allude to the question of multiple font handling. i.e. the other week there was a question in Fyne about displaying the check box characters U+2610 and U+2611. They are not in the NotoSans or NotoMono fonts but have been put in Noto-emoji. Should the font drawer handle this?

Reply:
We need to support whatever the user specifies in FYNE_FONT, or in the theme font handling.
Does that answer the question?

Reply:
Do we need to follow up on this message? Can you confirm the code works as expected when FYNE_FONT is specified on the command line?

Reply:
No, we do not need to follow up on the comment, it was copied from the original GO code just in case it was changed in the original code and someone was hunting for why Fyne is not behaving the same. 
Yes it does work with FYNE_FONT.